components/org.wso2.carbon.identity.application.authentication.handler.identifier/src/main/java/org/wso2/carbon/identity/application/authentication/handler/identifier/IdentifierHandler.java

```diff
@@ -397,6 +398,7 @@ protected void processAuthenticationResponse(HttpServletRequest request,
                 tenantAwareUsername = resolvedUserResult.getUser().getUsername();
                 username = UserCoreUtil.addTenantDomainToEntry(tenantAwareUsername, tenantDomain);
                 userId = resolvedUserResult.getUser().getUserID();
+                userStoreDomain = resolvedUserResult.getUser().getUserStoreDomain();
```

Comment:
Please verify a flow where we have the same username in both Asgardeo userstore and Default userstore?

Reply:
It filters with the tenant. So it will pick based on the relation to the tenant.